script/utils/GenerateConstantsJson.ts

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env ts-node
+
+import * as fs from "fs";
+import * as path from "path";
+
+// Import the constants from the TypeScript files
+import { CHAIN_IDs } from "../../utils/constants";
+import {
+  ZERO_ADDRESS,
+  USDC,
+  USDCe,
+  WETH,
+  WGHO,
+  QUOTE_TIME_BUFFER,
+  FILL_DEADLINE_BUFFER,
+  ARBITRUM_MAX_SUBMISSION_COST,
+  AZERO_GAS_PRICE,
+  CIRCLE_UNINITIALIZED_DOMAIN_ID,
+  ZK_L1_GAS_TO_L2_GAS_PER_PUBDATA_LIMIT,
+  ZK_L2_GAS_LIMIT,
+  ZK_MAX_GASPRICE,
+  L1_ADDRESS_MAP,
+  OP_STACK_ADDRESS_MAP,
+  L2_ADDRESS_MAP,
+  CIRCLE_DOMAIN_IDs,
+  OFT_EIDs,
+} from "../../deploy/consts";
+
+// Helper function to convert chain IDs object to the expected format
+function convertChainIdsToObject(chainIds: any): { [key: string]: number } {
+  const result: { [key: string]: number } = {};
+  for (const [key, value] of Object.entries(chainIds)) {
+    if (typeof value === "number") {
+      result[key] = value;
+    }
+  }
+  return result;
+}
+
+// Helper function to extract wrapped native tokens from TOKEN_SYMBOLS_MAP
+function extractWrappedNativeTokens(): { [key: string]: string } {
+  const result: { [key: string]: string } = {};
+
+  // Extract WETH addresses for each chain
+  for (const [chainId, addresses] of Object.entries(WETH)) {
+    const chainName = Object.keys(CHAIN_IDs).find(
+      (key) => CHAIN_IDs[key as keyof typeof CHAIN_IDs] === Number(chainId)
+    );
+    if (chainName) {
+      result[chainName] = addresses;
+    }
+  }
+
+  return result;
+}
+
+// Helper function to extract USDC addresses
+function extractUsdcAddresses(): { [key: string]: string } {
+  const result: { [key: string]: string } = {};
+
+  for (const [chainId, address] of Object.entries(USDC)) {
+    const chainName = Object.keys(CHAIN_IDs).find(
+      (key) => CHAIN_IDs[key as keyof typeof CHAIN_IDs] === Number(chainId)
+    );
+    if (chainName) {
+      result[chainName] = address;
+    }
+  }
+
+  return result;
+}
+
+// Helper function to extract USDCe addresses
+function extractUsdceAddresses(): { [key: string]: string } {
+  const result: { [key: string]: string } = {};
+
+  for (const [chainId, address] of Object.entries(USDCe)) {
```

Comment:
I don't think this is super reliable because the USDC symbol itself can be quite random. Zora has USDzC for example.